### PR TITLE
Upgrade to Rollup 3 and support hook handler

### LINKED ISSUE #12: Support Rollup 3 plugins with handler properties
The following fails as hooks can be objects with a handler property:
https://github.com/robertknight/rollup-cache/blob/baa76b829465b4f0da50b5ad168ee6df734f5522/lib/plugin.js#L74

Solution:
```js
const result = await (plugin.resolveId.handler || plugin.resolveId).call(this, id, importer, options);
```

It would need to be applied to all hooks.

### PR DESCRIPTION
**Pull request by dmnsgn**
This is a breaking change that probably needs testing: it seems to work the same for plugin caching but I have not tested prebuild.

Closes #12

### CODE REVIEW

**Comment by dmnsgn** [comment]
I have upgraded to yarn 3.6.0 (from v1) and reinstalled.

File: rollup.config.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import pkg from "./package.json";
+import pkg from "./package.json" assert { type: "json" };
```
Comment (robertknight) [inline rollup.config.js:1]: FYI this syntax is going to change in future. See https://github.com/tc39/proposal-import-attributes#synopsis.

**Review by robertknight — APPROVED** [review]
Thanks for this!

**Comment by robertknight** [comment]
This has been published in [v2.0.0](https://github.com/robertknight/rollup-cache/releases/tag/v2.0.0).

**Pull request merged**